vsphere/resource_vsphere_virtual_machine_test.go

```diff
@@ -13591,3 +13607,102 @@ resource "vsphere_virtual_machine" "vm" {
 		os.Getenv("VSPHERE_DATASTORE"),
 	)
 }
+
+func testAccResourceVSphereVirtualMachineReadVappChildResourcePool() string {
+	return fmt.Sprintf(`
+variable "dc" {default="%s"}
+variable "datastore" {default="%s"}
+variable "cluster" {default="%s"}
+variable "network" {default="%s"}
+variable "template" {default="%s"}
+variable "vm_name" {default="%s"}
+variable "disk_size" {default="%s"}
+variable "hostname" {default="%s"}
+variable "domain" {default="%s"}
+variable "ipv4address" {default="%s"}
+variable "ipv4cidrbits" {default="%s"}
+variable "ipv4gateway" {default="%s"}
+
+data "vsphere_datacenter" "dc" {
+  name = "${var.dc}"
+}
+
+data "vsphere_datastore" "datastore" {
+  name          = "${var.datastore}"
+  datacenter_id = "${data.vsphere_datacenter.dc.id}"
+}
+
+data "vsphere_compute_cluster" "cluster" {
+  name          = "${var.cluster}"
+  datacenter_id = "${data.vsphere_datacenter.dc.id}"
+}
+
+data "vsphere_network" "network" {
+  name          = "${var.network}"
+  datacenter_id = "${data.vsphere_datacenter.dc.id}"
+}
+
+data "vsphere_virtual_machine" "template" {
+  name          = "${var.template}"
+  datacenter_id = "${data.vsphere_datacenter.dc.id}"
+}
+
+data "vsphere_resource_pool" "pool" { 
+	name = "hashi-cluster1/Resources/hashi-vapp/ko-rp"
```

Comment:
Either make this a variable so and add it to the .devrc, or you could also have a resource_pool resource to create it on demand.